build.rs

```diff
@@ -63,8 +58,56 @@ fn configured_by_vcpkg() -> bool {
     false
 }
 
+fn static_or_not() {
+    use std::ascii::AsciiExt;
+
+    // On Windows-MSVC, always link dynamically
+    if cfg!(all(windows, target_env="msvc")) {
+        println!("cargo:rustc-link-lib=dylib=libpq");
+        return;
+    }
+
+    // Link unconditionally statically
+    if env::var_os("PQ_LIB_STATIC").is_some() {
+        println!("cargo:rustc-link-lib=static=pq");
+        return;
+    }
+
+    // Examine the per-target env vars
+    if let Ok(target) = env::var("TARGET") {
+        let pg_config_for_target = format!("PQ_LIB_STATIC_{}", target.to_ascii_uppercase().replace("-", "_"));
```

Comment:
This requires us adding a new

```rust
println!("cargo:rerun-if-env-changed={}", pg_config_for_target);
```

...line somewhere in the script. Also, you probably mean `pq_lib_static_for_target` here. :-)